src/llmtuner/data/template.py

```diff
@@ -514,6 +514,16 @@ def get_template_and_fix_tokenizer(name: str, tokenizer: "PreTrainedTokenizer")
 
 register_template(name="vanilla")
 
+register_template(
+    name="chatml_de",
+    format_user=StringFormatter(slots=["<|im_start|>user\n{{content}}<|im_end|>\n<|im_start|>assistant\n"]),
+    format_system=StringFormatter(slots=["<|im_start|>system\n{{content}}<|im_end|>\n"]),
+    format_separator=EmptyFormatter(slots=["\n"]),
+    default_system="Du bist ein freundlicher und hilfsbereiter KI-Assistent.",
+    stop_words=["<|im_end|>"],
+    replace_eos=True,
```

Comment:
the `replace_eos` option is unsafe when the stop word is not in the vocabulary with LLaMA-Factory's implementation, we would fix later, leave it as it is